engine/flink/management/src/main/scala/pl/touk/nussknacker/engine/management/FlinkRestManager.scala

```diff
@@ -226,6 +227,11 @@ class FlinkRestManager(config: FlinkConfig, modelData: ModelData, mainClassName:
         .body(program)
         .send()
         .flatMap(handleUnitResponse)
+        .recover({
+          //sometimes deploying takes too long, which causes TimeoutException while waiting for deploy response
+          //workaround for now, not the best solution though
+          case _: TimeoutException => Future.unit
```

Comment:
Please add logger.warn ... and maybe extend comment with some TODO of better solution like
`TODO: we should change logic of ManagementActor to always save action deploy (maybe we should rename it to events: deploy_attempt) and introduce new actions (events): deploy_success, deploy_failure(type of failure)`

Reply:
Done